script.py

```diff
@@ -0,0 +1,186 @@
+import requests
+import timeit
+import pyspark.sql.functions as f
+from pyspark.sql import SparkSession
+import argparse
+
+spark = SparkSession.builder.getOrCreate()
+
+program_description = '''
+This script retrieves the structures in complex with the molecules from parquet.
+'''
+
+parser = argparse.ArgumentParser(add_help=True, description=program_description)
+
+parser.add_argument('-p', '--parquet', help='Path to parquet folder', default=None, metavar='parquet_file_path', type=str, required=True)
+parser.add_argument('-u', '--unichem', help='Path to unichem src1src3.txt file', default=None, metavar='unichem_file_path', type=str, required=True)
+parser.add_argument('-e', '--ensembl', help='Path to pdb_chain_ensembl.csv file', default=None, metavar='ensembl_file_path', type=str, required=True)
+
+args = parser.parse_args()
+
+
+def get_structure(pdb_id):
+    """
+    Scrap the list of structures in complex with each molecule from PDBe API
+    Function to apply to a df column
+    --------------
+    Keyword arguments
+        pdb_id: id of the molecule for whom we want to know the structures associated
+    --------------
+    Return
+        data[pdb_id] : data is a dictionary, we want the value which is the list of structures
```

Comment:
I like the detailed docstring of this function. However writing a good docstring is always easier for a function which is easy to understand already. In this case the function is just a few lines. And its function is quite intuitive. Also the return value of this function is a list not a dictionary, or at least this is how it seems form the text. When writing documentation, always try to be short but very precise.

How I would re-phrase this docstring:
```python
def get_structure(pdb_compound_id: str) -> list:
    """Fetching structure identifiers from PDBkb REST API

    Args:
        pdb_compound_id: string, a single compound identifier
    Returns:
        List of PDB structure identifiers where the compound can be found
    """
```

What you see here:
* The input parameter's name is more telling. It's not just any identifier from PDB. This is a compound id.
* You can specify the type of the input parameters by using the `:` and the return type by `->`. This serves only documentation purposes. It's not enforced by anyhow.
* The docstring follows [google guidelines](https://google.github.io/styleguide/pyguide.html). In general, reading the style guide is a good time investment.